Release Notes/510.md

```diff
@@ -48,12 +48,22 @@
   - Issue: https://github.com/apple/swift-syntax/issues/2092
   - Pull Request: https://github.com/apple/swift-syntax/pull/2184
 
+- `IncrementalParseTransition`:
+  - Description: The initializer `IncrementalParseTransition.init(previousTree:edits:lookaheadRanges:reusedNodeCallback:)` is marked as deprecated. The deprecated initializer emits a warning indicating the use of `IncrementalParseTransition.init(previousIncrementalParseResult:edits:reusedNodeCallback:)`.
```

Comment:
Let’s write this a little bit more imperatively so that readers have an action to follow to resolve the deprecation.

```suggestion
  - Description: The initializer `IncrementalParseTransition.init(previousTree:edits:lookaheadRanges:reusedNodeCallback:)` is marked as deprecated. Use `IncrementalParseTransition.init(previousIncrementalParseResult:edits:reusedNodeCallback:)` instead.
```